crates/bevy_ecs/src/entity/mod.rs

```diff
@@ -954,12 +956,9 @@ impl Entities {
     #[inline]
     pub fn get(&self, entity: Entity) -> Option<EntityLocation> {
         if let Some(meta) = self.meta.get(entity.index() as usize) {
-            if meta.generation != entity.generation
-                || meta.location.archetype_id == ArchetypeId::INVALID
-            {
-                return None;
-            }
-            Some(meta.location)
+            (meta.generation == entity.generation)
+                .then_some(meta.location)
+                .flatten()
```

Comment:
`then_some().flatten()` is a little odd.  I think that might be `Option::filter()`?  

Hmm, or could this whole thing be replaced with `filter().and_then()` like

```rust
self.meta.get(entity.index() as usize)
    .filter(|meta| meta.generation == entity.generation)
    .and_then(|meta| meta.location)
```

?

(Sorry, I can't get GitHub to let me write that as a suggestion.)